src/Optimizer/Transformer/AmpStoryCssOptimizer.php

```diff
@@ -0,0 +1,244 @@
+        . '{function e(){document.documentElement.style.setProperty("--story-dvh",innerHeight/100+"px","important")}'
+        . 'addEventListener("resize",e,{passive:!0}),e()}';
+
+    /**
+     * Configuration store to use.
+     *
+     * @var TransformerConfiguration
+     */
+    private $configuration;
+
+    /**
+     * Instantiate a AmpStoryCssOptimizer object.
+     *
+     * @param TransformerConfiguration $configuration Configuration store to use.
+     */
+    public function __construct(TransformerConfiguration $configuration)
+    {
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * Apply transformations to the provided DOM document.
+     *
+     * @param Document        $document DOM document to apply the transformations to.
+     * @param ErrorCollection $errors   Collection of errors that are collected during transformation.
+     * @return void
+     */
+    public function transform(Document $document, ErrorCollection $errors)
+    {
+        if (!$this->configuration->get(AmpStoryCssOptimizerConfiguration::OPTIMIZE_AMP_STORY)) {
+            return;
+        }
+
+        $hasAmpStoryScript            = false;
+        $hasAmpStoryDvhPolyfillScript = false;
+        $styleAmpCustom               = null;
+
+        foreach ($document->head->childNodes as $childNode) {
+            if (! $childNode instanceof Element) {
+                continue;
+            }
+
+            if ($this->isAmpStoryScript($childNode)) {
+                $hasAmpStoryScript = true;
+                continue;
+            }
+
+            if ($this->isAmpStoryDvhPolyfillScript($childNode)) {
+                $hasAmpStoryDvhPolyfillScript = true;
+                continue;
+            }
+
+            if ($this->isStyleAmpCustom($childNode)) {
+                $styleAmpCustom = $childNode;
+                continue;
+            }
+        }
+
+        // We can return early if no amp-story script is found.
+        if (! $hasAmpStoryScript) {
+            return;
+        }
+
+        $this->appendAmpStoryCssLink($document);
+
+        if ($styleAmpCustom) {
+            $this->modifyAmpCustomCSS($styleAmpCustom);
+            // Make sure to not double install the dvh polyfill.
+            if (! $hasAmpStoryDvhPolyfillScript) {
+                $this->appendAmpStoryDvhPolyfillScript($document);
+            }
+        }
+
+        $this->supportsLandscapeSSR($document);
+        $this->aspectRatioSSR($document);
+    }
+
+    /**
+     * Check whether the element is an AMP Story element.
+     *
+     * @param Element $element Element to check.
+     * @return bool Whether the given element is an AMP story.
+     */
+    private function isAmpStoryScript(Element $element)
+    {
+        return ($element->tagName === Tag::SCRIPT)
+            && ($element->getAttribute(Attribute::CUSTOM_ELEMENT) === Extension::STORY);
```

Comment:
No need for the parentheses here (especially since the next method is similar and doesn't use them).
```suggestion
        return $element->tagName === Tag::SCRIPT
            && $element->getAttribute(Attribute::CUSTOM_ELEMENT) === Extension::STORY;
```